storage/migrations/6_tipset_processing_tables.go

```diff
@@ -0,0 +1,109 @@
+package migrations
+
+import (
+	"github.com/go-pg/migrations/v8"
+)
+
+// Schema version 6 merges tipset oriented processing tables
+
+func init() {
+
+	up := batch(`
+-- Merge visor_processing_statechanges and visor_processing_messages tables into a new tipset table
+ALTER TABLE public.visor_processing_statechanges RENAME TO visor_processing_tipsets;
+ALTER TABLE public.visor_processing_tipsets RENAME COLUMN claimed_until TO statechange_claimed_until;
+ALTER TABLE public.visor_processing_tipsets RENAME COLUMN completed_at TO statechange_completed_at;
+ALTER TABLE public.visor_processing_tipsets RENAME COLUMN errors_detected TO statechange_errors_detected;
+ALTER TABLE public.visor_processing_tipsets ADD COLUMN message_claimed_until timestamptz;
+ALTER TABLE public.visor_processing_tipsets ADD COLUMN message_completed_at timestamptz;
+ALTER TABLE public.visor_processing_tipsets ADD COLUMN message_errors_detected text;
+
+-- Copy data from visor_processing_messages before it is repurposed
+UPDATE public.visor_processing_tipsets t
+SET message_claimed_until = m.claimed_until,
+    message_completed_at = m.completed_at,
+    message_errors_detected = m.errors_detected
+FROM public.visor_processing_messages m
+WHERE m.tip_set = t.tip_set;
+
+
+-- Repurpose visor_processing_messages table
+DROP TABLE IF EXISTS public.visor_processing_messages;
+
+CREATE TABLE public.visor_processing_messages (
+	"cid" text NOT NULL,
+	"height" bigint,
+	"added_at" timestamptz NOT NULL,
+	"gas_outputs_claimed_until" timestamptz,
+	"gas_outputs_completed_at" timestamptz,
+	"gas_outputs_errors_detected" text,
+	PRIMARY KEY ("cid")
+);
+
+CREATE TABLE IF NOT EXISTS public.derived_gas_outputs (
+	"cid" text NOT NULL,
+	"from" text NOT NULL,
+	"to" text NOT NULL,
+	"value" text NOT NULL,
+	"gas_fee_cap" text NOT NULL,
+	"gas_premium" text NOT NULL,
+	"gas_limit" bigint,
+	"size_bytes" bigint,
+	"nonce" bigint,
+	"method" bigint,
+	"params" bytea,
+	"state_root" text NOT NULL,
+	"idx" bigint,
+	"exit_code" bigint,
+	"gas_used" bigint,
+	"return" bytea,
+	"parent_base_fee" text,
+	"base_fee_burn" text,
+	"over_estimation_burn" text,
+	"miner_penalty" text,
+	"miner_tip" text,
+	"refund" text,
+	"gas_refund" bigint,
+	"gas_burned" bigint,
+	PRIMARY KEY ("cid")
```

Comment:
Indexes to include here:
- `from` (using Hash)
- `to` (using Hash)
- `method`(using BTree)
- `exit_code` (using BTree)

Reply:
Done